app/static/js/index.js

```diff
@@ -0,0 +1,58 @@
+// Function to check if an element is in the viewport
+let inViewId = "";
+
+const inView = (element) => {
+  const rect = element.getBoundingClientRect();
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <=
+      (window.innerHeight || document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+};
+
+const updateNavigationView = (inViewId) => {
+  // Update the HTML content with the new value of inViewId
+  const navElements = document.querySelectorAll(".nav-element");
+  navElements.forEach((navElement) => {
+    const sectionId = navElement.getAttribute("data-section-id");
+
+    // Check if the sectionId matches the inViewId
+    if (sectionId === inViewId) {
+      console.log(sectionId);
+      // Apply the <strong> tag to the navigation link
+      const link = navElement.querySelector("a");
+      if (link) {
+        link.innerHTML = `<strong>${link.textContent}</strong>`;
+      }
+    } else {
+      // Remove <strong> tag if it exists
+      const link = navElement.querySelector("a");
+      if (link) {
+        link.innerHTML = link.textContent;
+      }
+    }
+  });
+};
+
+// Function to handle scroll events
+const handleScroll = () => {
+  const sections = document.querySelectorAll("section[id]");
```

Comment:
This is potentially a lot of work to do on every scroll event. Perhaps you’d improve the performance by moving this query selector out of the scroll handler?

Reply:
True that, it's gone